packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,339 @@
+import { stringify } from 'csv-stringify';
+import DataLoader from 'dataloader';
+import { Inject, Injectable, Scope } from 'graphql-modules';
+import { z } from 'zod';
+import { Session } from '../../auth/lib/authz';
+import { ClickHouse, sql } from '../../operations/providers/clickhouse-client';
+import { SqlValue } from '../../operations/providers/sql';
+import { OrganizationManager } from '../../organization/providers/organization-manager';
+import { Emails, mjml } from '../../shared/providers/emails';
+import { Logger } from '../../shared/providers/logger';
+import { S3_CONFIG, type S3Config } from '../../shared/providers/s3-config';
+import { auditLogSchema } from './audit-logs-types';
+
+const auditLogEventTypes = auditLogSchema.options.map(option => option.shape.eventType.value);
+
+const metadata = z.string().transform(value => {
+  try {
+    return {
+      ...JSON.parse(value),
+      ...auditLogSchema.parse(JSON.parse(value)),
+    };
+  } catch (e) {
+    return value;
+  }
+});
+
+export const AuditLogClickhouseObjectModel = z.object({
+  id: z.string(),
+  event_time: z.string(),
+  user_id: z.string(),
+  user_email: z.string(),
+  organization_id: z.string(),
+  event_action: z.enum(auditLogEventTypes as [string, ...string[]]),
+  metadata: metadata,
+});
+
+export type AuditLogType = z.infer<typeof AuditLogClickhouseObjectModel>;
+
+type AuditLogCSVType = {
+  ID: string;
+  Type: string;
+  CreatedAt: string;
+  UserID: string;
+  UserEmail: string;
+  OrganizationID: string;
+  Metadata: string;
+};
+
+const AuditLogClickhouseArrayModel = z.array(AuditLogClickhouseObjectModel);
+
+const AuditLogClickhouseCountModel = z.array(
+  z.object({
+    count: z.string(),
+  }),
+);
+
+@Injectable({
+  scope: Scope.Operation,
+})
+/**
+ * Responsible for accessing audit logs.
+ */
+export class AuditLogManager {
+  private logger: Logger;
+  private dataLoader = new DataLoader(async (organizationIds: readonly string[]) => {
+    const organizations = await Promise.all(
+      organizationIds.map(organizationId =>
+        this.organizationManager.getOrganization({ organizationId }),
+      ),
+    );
+    return organizations;
+  });
+
```

Comment:
TODO @TuvalSimha: remove this data loader, in the query result all logs will always belong to a single organization.